### PR TITLE
feat: api v2 change for solution plugin

### PR DESCRIPTION
**Pull request by jayzhang**

### CODE REVIEW

File: packages/api/src/v2/resourcePlugin.ts [hunk 1]
```diff
@@ -3,11 +3,13 @@
 
 import { Result } from "neverthrow";
 import { FxError, QTreeNode, TokenProvider, Void, Func, Json, Inputs } from "../index";
+import { EnvConfig } from "../schemas";
```
Comment (qinezh) [inline packages/api/src/v2/resourcePlugin.ts:6]: As we discussed, the schema of env config should be included in solution and shouldn't be exposed to API?

File: packages/api/src/v2/resourcePlugin.ts [hunk 2]
```diff
@@ -108,18 +110,17 @@ export interface ResourcePlugin {
...
+   * @returns {Json} part of {@link EnvProfile}
    */
   provisionResource?: (
     ctx: Context,
     inputs: Readonly<ProvisionInputs>,
-    provisionTemplate: Json,
+    envConfig: EnvConfig,
     tokenProvider: TokenProvider
-  ) => Promise<Result<ProvisionOutput, FxError>>;
+  ) => Promise<Result<Json, FxError>>;
```
Comment (1yefuwang1) [inline packages/api/src/v2/resourcePlugin.ts:123]: Perhaps return EnvProfile instead.

File: packages/api/src/v2/resourcePlugin.ts [hunk 3]
```diff
@@ -138,10 +139,10 @@ export interface ResourcePlugin {
   configureResource?: (
     ctx: Context,
     inputs: Readonly<ProvisionInputs>,
-    provisionOutput: Readonly<ProvisionOutput>,
-    provisionOutputOfOtherPlugins: Readonly<Record<PluginName, ProvisionOutput>>,
+    envConfig: EnvConfig,
+    envProfile: EnvProfile,
     tokenProvider: TokenProvider
-  ) => Promise<Result<ProvisionOutput, FxError>>;
+  ) => Promise<Result<Void, FxError>>;
```
Comment (1yefuwang1) [inline packages/api/src/v2/resourcePlugin.ts:145]: Returning Void means plugins are expected to mutate envConfig and envProfile. Consider returning a new copy instead.

**Review by hund030 — APPROVED** [review]

File: packages/api/src/v2/resourcePlugin.ts (hunk 3 above)
Comment (jayzhang) [reply packages/api/src/v2/resourcePlugin.ts:134]: If the API returns FxError, then core still need to persist the envProfile object, so just modify the input object makes the behavior more consistent.

**Review by KennethBWSong — APPROVED** [review]

**Review by YitongFeng-git — APPROVED** [review]
lgtm for bot/

**Review by XiaofuHuang — APPROVED** [review]

**Review by LongOddCode — APPROVED** [review]

**Review by jiayi-ruan — APPROVED** [review]
approve for spfx

**Review by kuojianlu — APPROVED** [review]
Approve for local debug part.

**Review by xzf0587 — APPROVED** [review]
approve

**Review by 1yefuwang1 — APPROVED** [review]

**Review by nliu-ms — APPROVED** [review]

**Review by qinezh — APPROVED** [review]

**Pull request merged**